doc/reference/s3gateway_api.md

```diff
@@ -18,21 +18,19 @@ Lists all of the branches across all of the repos as S3 buckets.
 
 ### Operations on Buckets
 
-Buckets are represented via `branch.repo`, e.g. the `master.images` bucket
-corresponds to the `master` branch of the `images` repo.
-
 #### DELETE Bucket
 
-Route: `DELETE /<branch>.<repo>/`.
+Route: `DELETE /<repo>/`.
 
-Deletes the branch. If it is the last branch in the repo, the repo is also
-deleted. Unlike S3, you can delete non-empty branches.
+Deletes the repo.
 
 #### GET Bucket (List Objects) Version 1
 
-Route: `GET /<branch>.<repo>/`
+Route: `GET /<repo>/?branch=<branch>`
 
-Only S3's list objects v1 is supported.
+Only S3's list objects v1 is supported. The `branch` parameter is a
+non-standard extension that allows you to specify which branch to on the given
+repo to list objects from; if unspecified, it defaults to `master`.
```

Comment:
Does this mean that it's effectively impossible to access non master branches using language clients? Or do they normally give you a way to inject random query parameters?

Reply:
The former. I'll try to think of a better scheme, but using the old system (`branch.repo` for the bucket name) in tandem with versioning didn't make much sense, because you could reference commits that were in an adjacent branch's history.